StortSpelProjekt/Engine/src/ECS/Components/GUI2DComponent.h

```diff
@@ -0,0 +1,78 @@
+#ifndef GUI2DCOMPONENT_H
+#define GUI2DCOMPONENT_H
+
+#include <map>
+#include <vector>
+#include "Component.h"
+class Window;
+class Texture;
+class Text;
+class Font;
+class Mesh;
+class Resource;
+class ShaderResourceView;
+class Events;
+class Eventbus;
+
+struct TextData;
+struct Vertex;
+struct ButtonPressed;
+
+namespace component
+{
+	class GUI2DComponent : public Component
+	{
+	public:
+		GUI2DComponent(Entity* parent);
+		virtual ~GUI2DComponent();
+
+		// Text functions
+		std::map<std::string, TextData>* const GetTextDataMap();
+		TextData* GetTextData(std::string name);
+
+		void AddText(std::string name);
+		void SubmitText(Text* text, std::string name);
+		void ReplaceText(Text* text, std::string name);
```

Comment:
Same here, also this got the same name as a macro defined in commondlg.h. This might cause problems, but it seems to work? :o

Reply:
Yes, ReplaceText should also be private. These functions are here because they needed to earlier in the development, but not anymore. And sure, I will rename it to be sure that it wont break anything